### PR TITLE
Make license package more resilient to unexpected output

### PR DESCRIPTION
**Pull request by puerco**
#### What type of PR is this?

/kind bug
/kind cleanup

#### What this PR does / why we need it:

This PR improves the licensing code to be more resilient when encountering unexpected license tags in the
classifier output. Since v2.0.0, [the classifier will now return a pseudo license tag when finding copyright information] in files. This caused our libraries to return an error when scanning files.

It also fixes another bug where the secondary license list was lost after scanning and classifying a file. 

The verbosity of the licensing code has now been reduced to make it more understandable.

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/release/issues/2729

#### Special notes for your reviewer:

/assign @jeremyrickard @xmudrii @cpanato 

#### Does this PR introduce a user-facing change?

```release-note
- Fixed a bug where the secondary license list returned by the classifier was not being returned
- Improved the licensing code to be more resilient to unexpected output from the classifier
- Licensing output is now less verbose. Use `--log-level=debug` to see all messages
```

### CODE REVIEW

File: pkg/license/implementation.go
Comment (cpanato) [inline pkg/license/implementation.go:58]: I was not sure if that was correct, i see now

**Review by cpanato — APPROVED** [review]
/lgtm

But i will add a 
/hold 

Would like to run some manual test
But if you already ran feel.free to lift

**Comment by puerco** [comment]
The CLI runs. It should run in `krel` too as we're not changing anything that affects it. But I can only test it once this merges and we can run a build using someones branch

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Comment by puerco** [comment]
@cpanato let me know when we can lift the hold to test krel :rocket:

**Comment by puerco** [comment]
@cpanato let me know when we can lift the hold to test krel :rocket:

**Comment by puerco** [comment]
@cpanato let me know when we can lift the hold to test krel :rocket:

**Comment by cpanato** [comment]
doing the update in krel and testing

**Pull request merged**
